### PR TITLE
Use watchbot-progress

### PR DESCRIPTION
**Pull request by emilymdubois**
Refs https://github.com/mapbox/ecs-watchbot/issues/71 aftermath

We extricated the progress functions into their own [`watchbot-progress` module](https://github.com/mapbox/watchbot-progress). This PR scraps the existing progress logic and documentation, and adds `watchbot-progress` as a dependency.

cc @rclark

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -15,7 +15,6 @@ module.exports = {
   log: require('./lib/logs').log,
   logStream: require('./lib/logs').logStream,
   fetch: require('./lib/logs').fetch,
-  progress: require('./lib/progress'),
```
Comment (rclark) [inline index.js]: I think you should still export this so folks can continue to `require('watchbot').progress()` like they're used to.

File: package.json [hunk 2]
```diff
@@ -24,8 +24,7 @@
     "docs": "documentation build --format html --output docs --config docs/config.json && opener docs/index.html"
   },
   "bin": {
-    "watchbot-log": "./bin/watchbot-log.js",
-    "watchbot-progress": "./bin/watchbot-progress.js"
```
Comment (rclark) [inline package.json]: Similarly, we'd still like to be able to

```
$ npm link watchbot
$ watchbot-progress status yadda
```

File: readme.md [hunk 3]
```diff
@@ -246,7 +246,12 @@ By setting the `Reduce` parameter to true, Watchbot will be capable of helping t
 progress of distributed map-reduce operations. This is useful if your stack performs
 a bunch of individual jobs that need to be "rolled up" into a final output of some sort.
 
-See the included `reduce-examples.md` for some basic worker recipes.
+Watchbot uses the `watchbot-progress` dependency for reporting progress to Watchbot
+as part of the messaging flow below. By enabling reduce-mode, Watchbot will initialize
+a DynamoDB progress table and configure necessary permissions for you.
+
+For examples and documentation on the available progress functions, see the
+[watchbot-progress module](https://github.com/mapbox/watchbot-progress).
```
Comment (rclark) [inline readme.md]: I would say we want to present the progress tracker as something that's bundled into watchbot for you -- not something you should run out and fetch for yourself. To me that means you'd still include instructions here on how to install the scripts on your container, but definitely link over to the new repository for instruction on how to actually use them at runtime.

**Comment by emilymdubois** [comment]
Thanks @rclark! Added some of that material back.

Is [this](https://github.com/mapbox/ecs-watchbot/blob/533d0b39b600b832dd484def7aeccc5c7a621e2c/package.json#L28) the best way to require resources from a dependency in the package.json?

**Comment by rclark** [comment]
@emilymdubois this LGTM.

**Pull request merged**